editor.planx.uk/src/pages/Team/components/AddFlow/index.tsx

```diff
@@ -0,0 +1,144 @@
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import Dialog, { dialogClasses } from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogTitle from "@mui/material/DialogTitle";
+import { styled } from "@mui/material/styles";
+import Typography from "@mui/material/Typography";
+import { isAxiosError } from "axios";
+import { Form, Formik, FormikConfig } from "formik";
+import React, { useState } from "react";
+import { useNavigation } from "react-navi";
+import { AddButton } from "ui/editor/AddButton";
+
+import { useStore } from "../../../FlowEditor/lib/store";
+import { BaseFormSection } from "./BaseFormSection";
+import { CreateFromCopyFormSection } from "./CreateFromCopyFormSection";
+import { CreateFromTemplateFormSection } from "./CreateFromTemplateFormSection";
+import { CreateFlow, validationSchema } from "./types";
+
+// TODO: Standardise or share this
+export const StyledDialog = styled(Dialog)(({ theme }) => ({
+  [`& .${dialogClasses.paper}`]: {
+    width: "100%",
+    maxWidth: theme.breakpoints.values.md,
+    borderRadius: 0,
+    borderTop: `20px solid ${theme.palette.primary.main}`,
+    background: theme.palette.background.paper,
+    margin: theme.spacing(2),
+  },
+}));
+
+export const AddFlow: React.FC = () => {
+  const { navigate } = useNavigation();
+  const { teamId, createFlow, teamSlug } = useStore();
+
+  const initialValues: CreateFlow = {
+    mode: "new",
+    flow: {
+      slug: "",
+      name: "",
+      sourceId: "",
+    },
+  };
+
+  const onSubmit: FormikConfig<CreateFlow>["onSubmit"] = async (
+    { mode, flow },
+    { setFieldError },
+  ) => {
+    let newFlowId: string | undefined;
+
+    try {
+      switch (mode) {
+        case "new":
+          newFlowId = await createFlow(teamId, flow.slug, flow.name);
+          break;
+        case "copy":
+          // newFlowId = await createFlowFromCopy(flow);
+          break;
+        case "template":
+          // newFlowId = await createFlowFromTemplate(flow);
+          break;
```

Comment:
Intentionally commented out as these options are disabled, leaving here as an indication of what will come next.